### PR TITLE
Fix testShrinkAfterRollover

### LINKED ISSUE #68395: [CI] TimeSeriesDataStreamsIT.testShrinkAfterRollover
**Build scan**:

https://gradle-enterprise.elastic.co/s/nqwhqniws3pp2

**Repro line**:

```
./gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.TimeSeriesDataStreamsIT.testShrinkAfterRollover" -Dtests.seed=1DCA2402DADCCD33 -Dtests.security.manager=true -Dtests.locale=es-AR -Dtests.timezone=America/Grand_Turk -Druntime.java=11
```

**Reproduces locally?**:

Not exactly - it failed for me locally with `the rollover action created the rollover index`.

**Applicable branches**:

`master`

**Failure history**:

https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/London&tests.container=org.elasticsearch.xpack.ilm.TimeSeriesDataStreamsIT&tests.sortField=FAILED&tests.test=testShrinkAfterRollover&tests.unstableOnly=true

**Failure excerpt**:

```
java.lang.AssertionError: the shrunken index was deleted by the delete action
at org.elasticsearch.xpack.ilm.TimeSeriesDataStreamsIT.lambda$testShrinkAfterRollover$12(TimeSeriesDataStreamsIT.java:127)
at org.elasticsearch.xpack.ilm.TimeSeriesDataStreamsIT.testShrinkAfterRollover(TimeSeriesDataStreamsIT.java:127)
```

### PR DESCRIPTION
**Pull request by andreidan**
This fixes the flakiness in testShrinkAfterRollover which was introduce by the
restrictive `allocate` action in the `cold` phase. This changes the full policy
we use in the ILM ITs to use an `allocate` action in the `cold` phase that includes
all nodes.

This commmit also improves the logging in TimeSeriesDataStreamsIT and other ILM actions
ITs.

Fixes #68395

### CODE REVIEW

File: x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/TimeSeriesRestDriver.java [hunk 1]
```diff
@@ -177,7 +177,8 @@ public static void createFullPolicy(RestClient client, String policyName, TimeVa
         warmActions.put(ShrinkAction.NAME, new ShrinkAction(1, null));
         Map<String, LifecycleAction> coldActions = new HashMap<>();
         coldActions.put(SetPriorityAction.NAME, new SetPriorityAction(0));
-        coldActions.put(AllocateAction.NAME, new AllocateAction(0, singletonMap("_name", "javaRestTest-3"), null, null));
+        coldActions.put(AllocateAction.NAME, new AllocateAction(0, singletonMap("_name", "javaRestTest-0,javaRestTest-1,javaRestTest-2," +
```
Comment (andreidan) [inline x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/TimeSeriesRestDriver.java:180]: this is where the fix for the test flakiness is (the rest of this PR is meant to aid with debugging test failures)

**Review by dakrone — APPROVED** [review]
LGTM, thanks for working on these Andrei

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

**Pull request merged**